R/02-templating-class.R

```diff
@@ -0,0 +1,26 @@
+#' renderCustomFragments
+#' 
+#' Peform the rendering of subfragments for a given report, should return a list of html fragments to 
+#' be available to inject into mustach templates
+#'
+#' @param reportData typed report json, the type should be the name of the report
+#' @return vector of rendered HTML fragments that will be included in the data available to the final report template. The fragments will be
+#' available to the templates as renderedFragments.html
+#' 
+#' @export
+renderCustomFragments <- function(reportData) 0
```

Comment:
Yeah, this was suggested as the base function. Without it the setMethod/setGeneric complains that there isn't a function yet defined.